google-cloud-clients/google-cloud-firestore/src/test/java/com/google/cloud/firestore/LocalFirestoreHelper.java

```diff
@@ -121,6 +126,52 @@
 
   public static final Precondition UPDATE_PRECONDITION;
 
+  public static class CompositeField {
+
+    public String stringField;
+    public PrimitiveField primitiveField;
+
+    public CompositeField(String stringField, PrimitiveField primitiveField) {
+      this.stringField = stringField;
+      this.primitiveField = primitiveField;
+    }
+
+    @Override
+    public boolean equals(Object o) {
+      if (this == o) {
+        return true;
+      }
+      if (o == null || getClass() != o.getClass()) {
+        return false;
+      }
+      CompositeField that = (CompositeField) o;
+      return stringField.equals(that.stringField) && primitiveField.equals(that.primitiveField);
+    }
+  }
+
+  public static class PrimitiveField {
```

Comment:
Would it be possible to use the existing SingleField class instead of this new helper?